### ISSUE TITLE
Writing and reading offsets only from  kafka  

### ISSUE THREAD

**Issue by naresh-kotha-ck** [pinterest/secor#1647]
Hi Team 

I have noticed that offsets are still being written to zookeeper even if I specify below config variables 

kafka.offsets.storage=kafka

kafka.dual.commit.enabled= false

Is this expected or am I doing some thing wrong.

**Comment by naresh-kotha-ck** [comment]
we are using kafka 2.5

**Comment by HenryCaiHaiying** [comment]
Secor use ZK for 2 purposes:

1. keep track of kafka message progress per topic/partition
2. use ZK as a distributed lock during file uploading

Purpose 1 can be switched to kafka as the offset storage, but the
dependency on ZK as a global lock is still there.

**Comment by naresh-kotha-ck** [comment]
yes but for purpose 1 I still see offsets in zookeeper even I set below properties 

kafka.offsets.storage=kafka

kafka.dual.commit.enabled= false

Can you please let me know if I am missing it

**Comment by HenryCaiHaiying** [comment]
For purpose 1, I think there are still some code that writes the offset to
ZK (while also writing to kafka), but the reader is only reading offsets
from Kafka.  If you prefer, you can submit a PR to clear those writing
places (I think they are mostly in Uploader.java).

**Comment by naresh-kotha-ck** [comment]
Sure I will create a PR for that . Thanks for the details

**Comment by DomWos** [comment]
Hey @HenryCaiHaiying, 
I also think there is a bug or inconsistency in `SecorKafkaMessageIterator` or I don't understand something correctly. 
Currently, there is a variable that is responsible for skipping reading offsets from zookeeper, but the variable is equal to 
`offsetStorage.equals("kafka") && dualCommitEnabled.equals("true")`. So, if I understand correctly, if we set `offset.storage=kafka` and `dual.commit.enabled=false` we are going to try to read offsets from Zookeeper, even though they will never be there due to `dual.commit.enabled=false`. Shouldn't the check be for `false` instead of `true` ?

**Comment by HenryCaiHaiying** [comment]
I think 'dual.commit.enabled' = false, the offsets will be written to ZK as
default.

**Comment by DomWos** [comment]
Yeah, that is correct for the old kafka consumer used by legacy iterator. However, in the `SecorKafkaMessageIterator` the new kafka consumer is used, which doesn't really even have the `dual.commit.enabled` property available, nor it's actually set by Secor. So, in the new consumer we won't ever save the offsets to Zookeeper, but for some reason Secor will always try to read them from there.

**Comment by dpavlov-smartling** [comment]
Hello @HenryCaiHaiying  and @naresh-kotha-ck,
Is there any updates and/or clarification of this situation?
Since Kafka 2.0 all internal Kafka tools don't support interaction with offsets in Zookeeper, so it is very hard to work with them if you need to reset offset, etc. 
If it possible to confirm that with configuration
```
kafka.offsets.storage=kafka
kafka.dual.commit.enabled=true
```
Secor will read offsets only from Kafka storage? I do understand that it will write to Zookeeper as well, but we want to migrate from Zookeeper offset storage to Kafka and want to be sure that we understand how Secor will interact after restart with updated configuration.

**Comment by HenryCaiHaiying** [comment]
By looking at the code (SecorConsumerRebalanceListener.java), it looks like it will skip zookeeper when the Kafka.offsets.storage=kafka:

```
    public void onPartitionsAssigned(Collection<TopicPartition> collection) {
        if (skipZookeeperOffsetSeek) {
            LOG.debug("offset storage set to kafka. Skipping reading offsets from zookeeper");
            return;
        }
        Map<TopicPartition, Long> committedOffsets = getCommittedOffsets(collection);

```
